### PR TITLE
[ESIMD] Doxygen update IV. simd_view/impl,...

### PR DESCRIPTION
**Pull request by kbobrovs**
Signed-off-by: Konstantin S Bobrovsky <konstantin.s.bobrovsky@intel.com>

### CODE REVIEW

**Review by bader — APPROVED** [review]
sycl/doc/doxygen.cfg.in looks okay.

**Comment by kbobrovs** [comment]
The Windows-only failure is unrelated:
[2022-02-06T08:56:05.043Z]   SYCL :: USM/hmemllaligned.cpp

File: sycl/include/sycl/ext/intel/experimental/esimd/detail/simd_view_impl.hpp [hunk 1]
```diff
@@ -25,21 +25,40 @@ namespace detail {
 /// @addtogroup sycl_esimd_core
 /// @{
 
-/// The simd_view base class.
-/// It is an internal class implementing basic functionality of simd_view.
+/// Base class for "simd view" types.
+/// It is an internal class implementing basic functionality inherited by all
+/// simd_view specializations. Objects of this type and its subclasses provide a
+/// a "view" into objects of types inheriting from simd_obj_impl - @ref simd or
```
Comment (kychendev) [inline sycl/include/sycl/ext/intel/experimental/esimd/detail/simd_view_impl.hpp:31]: Typo with repeated "a".

File: sycl/include/sycl/ext/intel/experimental/esimd/detail/simd_view_impl.hpp [hunk 2]
```diff
@@ -71,14 +97,20 @@ class simd_view_impl {
...
   /// Default copy constructor.
   simd_view_impl(const simd_view_impl &Other) = default;
 
   /// Default move constructor.
   simd_view_impl(simd_view_impl &&Other) = default;
 
-  /// Implicit conversion to simd type.
+  /// Implicit conversion to a simd object with potentially different element
+  /// type. Reads the viewed region from the target, converts elements to the
+  /// requested type and returns as a simd object. Available only then the type
```
Comment (kychendev) [inline sycl/include/sycl/ext/intel/experimental/esimd/detail/simd_view_impl.hpp:111]: Typo: "then" -> "when"

**Comment by kbobrovs** [comment]
@kychendev - thanks, fixed. Please take another look.

**Review by kychendev — APPROVED** [review]

**Comment by kbobrovs** [comment]
@bader, could you please re-approve? sycl/doc/doxygen.cfg.in did not change in the latest commit.

**Pull request merged**